src/data.rs

```diff
@@ -0,0 +1,107 @@
+use core::fmt;
+
+use rand::{thread_rng, Rng};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct Reading {
+    temperature: f64,
+    humidity: f64,
+    pressure: i32,
+    dust_concentration: f64,
+    air_purity: String,
+}
+
+impl Reading {
+    pub(crate) fn new(
+        temperature: f64,
+        humidity: f64,
+        pressure: i32,
+        dust_concentration: f64,
+        air_purity: String,
+    ) -> Reading {
+        Reading {
+            temperature,
+            humidity,
+            pressure,
+            dust_concentration,
+            air_purity,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub(crate) enum AirPurity {
+    Dangerous,
+    High,
+    Low,
+    FreshAir,
+}
+
+impl AirPurity {
+    pub(crate) fn from_value(value: f64) -> AirPurity {
+        match value {
+            value if (f64::MIN..=50.0).contains(&value) => AirPurity::FreshAir,
```

Comment:
I wonder if just using the same pattern you've used lower 

```suggestion
            value if value < 50.0 => AirPurity::FreshAir,
```
Would be more performant ? Not sure if this would need to check the whole range here or not. In any case keeping consistency in the pattern might be nice. Not a huge deal by any means.